homeassistant/components/kodi/media_player.py

```diff
@@ -370,27 +371,42 @@ async def async_added_to_hass(self):
             return
 
         if self._connection.connected:
-            self._on_ws_connected()
-
-        self.async_on_remove(
-            async_track_time_interval(
-                self.hass,
-                self._async_connect_websocket_if_disconnected,
-                WEBSOCKET_WATCHDOG_INTERVAL,
+            await self._on_ws_connected()
+
+        def start_watchdog(event=None):
+            """Start websocket watchdog."""
```

Comment:
```suggestion
        async def start_websocket(event=None):
            """Start websocket."""
            await self._async_connect_websocket_if_disconnected()
```

Reply:
While this may get the initial state 10 seconds (`WEBSOCKET_WATCHDOG_INTERVAL`) faster if kodi is available and responding, it will block HA startup longer than necessary if the kodi instance does not respond.
In my opinion, it's better to wait a bit and let HA start up before making the first connection attempt.

Reply:
Since its not happing until after the start event and its async, it shouldn't block anything since I/O is being awaited.

What am I missing here?

Reply:
My mistake, it should indeed be possible to connect right away. I will make the change soon.